File: python_sieve/sieve.py
import math;
import time;

def find_primes(max):
    bv = [True] * ((max // 2) + 1)
    bv[0] = False;

    for i in range(3,1+int(math.sqrt(float(max))), 2):
        #print(i)
        if bv[i // 2]:
            j = i
            while i * j < max:
                bv[(i * j) // 2] = False
                j += 2
            #print("Prime")

    numPrimes = bv.count(True)
    
    if numPrimes != 78498:
        print(f"{numPrimes} != 78498")


start = time.time()
iterations = 0

while (time.time() - start <= 5):
    find_primes(1000000)
    iterations += 1

print(f"Python did {iterations} iterations in 5 seconds")

File: lua_sieve/sieve.lua
local max = 1000000
local start = os.time()
local iterations = 0
local diff = os.difftime(os.time(), start)


local function find_primes (max_num)
    local bv = {};
    local arr_len = math.floor(max_num / 2) + 1
    for i = 1, arr_len, 1 do
        bv[i] = true
    end

    for i = 3, math.floor(math.sqrt(max_num)), 2 do
        local ch = math.floor(i / 2)

        if bv[ch] then
            local j = i
            while i * j < max_num do
                bv[math.floor((i * j) / 2)] = false
                j = j + 2
            end
        end
    end

    local num = -1;
    for _, p in pairs(bv) do
        if p then
            num = num + 1
        end
    end

    return num
end

while diff < 5 do
    find_primes(max)
    iterations = iterations + 1
    diff = os.difftime(os.time(), start)
end
print(string.format("Lua did %i iterations in 5 seconds", iterations))
